src/os/CMakeLists.txt

```diff
@@ -141,6 +137,32 @@ if(WITH_SPDK)
   endforeach()
 endif(WITH_SPDK)
 
+if(WITH_PMEM)
+  include(ExternalProject)
+  ExternalProject_Add(nvml_ext
+    DOWNLOAD_DIR ${CMAKE_BINARY_DIR}/src/
+    GIT_REPOSITORY "https://github.com/ceph/nvml.git"
+    GIT_TAG  "dd5b62dbc2cbbe048087dea"
+    SOURCE_DIR ${CMAKE_BINARY_DIR}/src/nvml
+    CONFIGURE_COMMAND ""
+    BUILD_COMMAND $(MAKE)
+    BUILD_IN_SOURCE 1
+    INSTALL_COMMAND "true")
+
+  ExternalProject_Add_Step(nvml_ext forcebuild
+    DEPENDEES configure
+    DEPENDERS build
+    COMMAND "true"
+    ALWAYS 1)
+
+  add_library(pmem STATIC IMPORTED GLOBAL)
+  add_dependencies(pmem nvml_ext)
+  target_link_libraries(os  pmem)
+  include_directories("${CMAKE_BINARY_DIR}/src/nvml/src/include")
```

Comment:
instead of applying this `include` globally, i'd suggest add this include directory only when compiling `libos`, like:
```diff
--- a/src/os/CMakeLists.txt
+++ b/src/os/CMakeLists.txt
@@ -157,10 +157,10 @@ if(WITH_PMEM)

   add_library(pmem STATIC IMPORTED GLOBAL)
   add_dependencies(pmem nvml_ext)
-  target_link_libraries(os  pmem)
-  include_directories("${CMAKE_BINARY_DIR}/src/nvml/src/include")
   set_target_properties(pmem PROPERTIES
     IMPORTED_LOCATION "${CMAKE_BINARY_DIR}/src/nvml/src/nondebug/libpmem.a")
+  target_link_libraries(os pmem)
+  target_include_directories(os PRIVATE "${CMAKE_BINARY_DIR}/src/nvml/src/include")
 endif(WITH_PMEM)

 if(WITH_LTTNG AND WITH_EVENTTRACE)
```